tests/ui/codegen/equal-pointers-unequal/as-cast/inline2.rs

```diff
@@ -0,0 +1,33 @@
+//@ known-bug: #107975
+//@ compile-flags: -Copt-level=2
+//@ run-pass
+//@ check-run-results
+//@ normalize-stdout-test: "\d+" -> "<..>"
+
+// FIXME(#127914)
+//@ normalize-stdout-test: "uploaded .* waiting for result\n" -> ""
```

Comment:
Is 'println' the only way makes the value of the pointer change (showing the effect in #107975)? Can we influence the compiler with other statements, adding 'assert_eq' for testing, rather than relying on the output of stdout?

Reply:
We can just do `assert_eq!(format!(..), ..)`, but that won't have the same normalization applied as in `check-run-results`. Should I try to change this or is stdout fine?

Reply:
I mean in this example:
```
    println!("{a:?} == {b:?} -> ==: {}, cmp_in: {}, cmp: {}", a == b, cmp_in(a, b), cmp(a, b));
    assert_eq!(a.to_string(), b.to_string());
```
Could we change it to
```
    assert_ne!(a, b);
    assert!(!cmp_in(a, b));
    assert!(cmp(a, b));
    assert_eq!(a.to_string(), b.to_string());
```
After all, in the tests, we don't care about the output of a and b in stdout.

Reply:
And yea I think `assert_eq!(format!(..), ..)` is a better way if we could get rid of all the `.stdout` files